web/src/app/admin/AdminConfig.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useState } from 'react'
+import { useQuery } from '@apollo/react-hooks'
+import Button from '@material-ui/core/Button'
+import Card from '@material-ui/core/Card'
+import Grid from '@material-ui/core/Grid'
+import Typography from '@material-ui/core/Typography'
+import { makeStyles } from '@material-ui/core/styles'
+import gql from 'graphql-tag'
+import { startCase, isEmpty, uniq } from 'lodash-es'
+import chain from 'lodash'
+import AdminSection from './AdminSection'
+import AdminDialog from './AdminDialog'
+import PageActions from '../util/PageActions'
+import { Form } from '../forms'
+import { InputAdornment, TextField } from '@material-ui/core'
+import CopyText from '../util/CopyText'
+import Spinner from '../loading/components/Spinner'
+import { GenericError } from '../error-pages'
+
+const query = gql`
+  query getConfig {
+    config(all: true) {
+      id
+      description
+      password
+      type
+      value
+    }
+    configHints {
+      id
+      value
+    }
+  }
+`
+const mutation = gql`
+  mutation($input: [ConfigValueInput!]) {
+    setConfig(input: $input)
+  }
+`
+
+const useStyles = makeStyles((theme) => ({
+  gridContainer: {
+    [theme.breakpoints.up('md')]: {
+      justifyContent: 'center',
+    },
+  },
+  gridItem: {
+    [theme.breakpoints.up('md')]: {
+      maxWidth: '65%',
+    },
+  },
+  groupTitle: {
+    fontSize: '1.1rem',
+  },
+  saveDisabled: {
+    color: 'rgba(255, 255, 255, 0.5)',
+  },
+}))
+
+interface ConfigValues {
+  [id: string]: string
+}
+
+export default function AdminConfig(): JSX.Element {
```

Comment:
we should be able to leave out the `JSX.Element` return types, I believe that is inferred (we haven't had a need to do that elsewhere)

Reply:
This is required not because of the typescript-eslint rules, all .tsx files have it now but it was just added when the eslint-explicit-function-return-type was added.

Reply:
Ah I see, new eslint rule. Gotcha 👍